packages/zwave-js/src/lib/commandclass/PowerlevelCC.ts

```diff
@@ -0,0 +1,409 @@
+
+	public async getPowerlevel(): Promise<
+		Pick<PowerlevelCCReport, "powerlevel" | "timeout"> | undefined
+	> {
+		this.assertSupportsCommand(PowerlevelCommand, PowerlevelCommand.Get);
+
+		const cc = new PowerlevelCCGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+		});
+		const response = await this.driver.sendCommand<PowerlevelCCReport>(
+			cc,
+			this.commandOptions,
+		);
+		if (response) {
+			return pick(response, ["powerlevel", "timeout"]);
+		}
+	}
+
+	public async startTestNode(
+		testNodeId: number,
+		powerlevel: Powerlevel,
+		testFrameCount: number,
+	): Promise<void> {
+		this.assertSupportsCommand(
+			PowerlevelCommand,
+			PowerlevelCommand.TestNodeSet,
+		);
+
+		const cc = new PowerlevelCCTestNodeSet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			testNodeId,
+			powerlevel,
+			testFrameCount,
+		});
+		await this.driver.sendCommand(cc, this.commandOptions);
+	}
+
+	public async getTestNodeResult(): Promise<
+		| Pick<
+				PowerlevelCCTestNodeReport,
+				"testNodeId" | "testStatus" | "testFrameAcknowlegedCount"
+		  >
+		| undefined
+	> {
+		this.assertSupportsCommand(
+			PowerlevelCommand,
+			PowerlevelCommand.TestNodeGet,
+		);
+
+		const cc = new PowerlevelCCTestNodeGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+		});
+		const response = await this.driver.sendCommand<PowerlevelCCTestNodeReport>(
+			cc,
+			this.commandOptions,
+		);
+		if (response) {
+			return pick(response, [
+				"testNodeId",
+				"testStatus",
+				"testFrameAcknowlegedCount",
+			]);
+		}
+	}
+}
+
+@commandClass(CommandClasses.Powerlevel)
+@implementedVersion(1)
+export class PowerlevelCC extends CommandClass {
+	declare ccCommand: PowerlevelCommand;
+}
+
+type PowerlevelCCSetOptions = CCCommandOptions &
+	(
+		| {
+				powerlevel: Powerlevel;
+				timeout: number;
+		  }
+		| {
+				powerlevel: typeof Powerlevel.NormalPower;
+		  }
+	);
+
+@CCCommand(PowerlevelCommand.Set)
+export class PowerlevelCCSet extends PowerlevelCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | PowerlevelCCSetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			if ("timeout" in options) {
+				if (options.timeout < 1 || options.timeout > 255) {
+					throw new ZWaveError(
+						`${this.constructor.name}: The timeout parameter must be between 1 and 255.`,
+						ZWaveErrorCodes.Argument_Invalid,
+					);
+				}
+				this.timeout = options.timeout;
+			}
+			this.powerlevel = options.powerlevel;
+		}
+	}
+
+	public powerlevel: Powerlevel;
+	public timeout?: number;
+
+	public serialize(): Buffer {
+		this.payload = Buffer.from([this.powerlevel, this.timeout ?? 0x00]);
```

Comment:
Just thinking out of the box, since I've been burned a few times.
If valid values for the timeout are ranging from 1 to 255 and the timeout must be ignored for "normal power", it could theoretically be that some weird device chokes on a timeout value of 0.
So to be sure, lets make 1 the default.